valinor/main.py

```diff
@@ -98,46 +101,46 @@ def main():
     # pass empty data to the tool for things we don't care about when just
     # debugging (in the future we could add source files by reading the debug
     # info from the file being debugged)
-    data = {
-        'name': file_base_name,     # project name
-        'core': '',                 # core
-        'linker_file': '',          # linker command file
-        'include_paths': [],        # include paths
-        'source_paths': [],         # source paths
-        'source_files_c': [],       # c source files
-        'source_files_cpp': [],     # c++ source files
-        'source_files_s': [],       # assembly source files
-        'source_files_obj': [],     # object files
-        'source_files_lib': [],     # libraries
-        'macros': [],               # macros (defines)
-        'project_dir': {
-            'name': '.' + os.path.sep,
-            'path' : projectfile_dir
-        },
-        'output_dir': os.path.relpath(executable_dir, projectfile_dir) + os.path.sep,
-        'misc': [],
-        'target': args.target
+    project_data = {
+        'common': {
+            'target': [args.target],  # target
+            'build_dir': [projectfile_dir],
+            'debugger': ['cmsis-dap'],   # TODO: find out what debugger is connected
+            'linker_file': ['None'],
+            'export_dir': [projectfile_dir],
+            'output_dir': {
+                'rel_path' : [''],
+                'path' : [os.path.relpath(executable_dir, projectfile_dir) + os.path.sep],
+            }
+        }
     }
 
-    exporter = tool.ToolsSupported().get_value(ide_tool, 'exporter')
-    # generate debug project files (if necessary)
-    projectfile_path, projectfiles = tool.export(exporter, data, ide_tool, project_settings)
-    if projectfile_path is None:
-        logging.error("failed to generate project files")
-        sys.exit(1)
+    projects = {
+        'projects' : {}
+    }
+    project = Project(file_base_name, [project_data], PgenWorkspace(projects))
+    project.export(ide_tool, False)
 
     # perform any modifications to the executable itself that are necessary to
     # debug it (for example, to debug an ELF with Keil uVision, it must be
     # renamed to have the .axf extension)
-    executable = tool.fixup_executable(exporter, args.executable, ide_tool)
+    executable = args.executable
+    if ide_tool == 'uvision':
```

Comment:
I removed fix up executable as it did not make sense to have it in Project and Tools, it's just needed for uvision so I placed it here.